gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifyContentsPerExecutor.java

```diff
@@ -62,13 +64,13 @@ public IdentifyContentsPerExecutor(GCParams gcParams) {
             bloomFilterSize);
   }
 
-  protected Function<Reference, IdentifiedResult> computeExpiredContentsFunc(
+  protected Function<String, IdentifiedResult> computeExpiredContentsFunc(
```

Comment:
shouldn't the type still stay `Reference` rather than using the serialized JSON string?

Reply:
This Lambda is passed from driver to executor, hence spark needs JAVA serialization if we keep `Reference`.
As described in the PR description, wanted to avoid java serialization.

Once received in executor, deserializing it in the method that this Lamdba calls.